hazelcast/src/main/java/com/hazelcast/client/config/ClientSqlConfig.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.internal.util.Preconditions;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+
+/**
+ * Contains the SQL configuration for the client.
+ *
+ * @since 5.2
+ */
+public class ClientSqlConfig {
+    private ClientSqlResubmissionMode resubmissionMode = ClientSqlResubmissionMode.NEVER;
+
+    /**
+     * Sets the resubmission mode for failing SQL queries. See {@link
+     * ClientSqlResubmissionMode}. The default value is {@link
+     * ClientSqlResubmissionMode#NEVER NEVER}.
+     *
+     * @since 5.2
+     */
+    @Nonnull
+    public ClientSqlConfig setSqlResubmissionMode(ClientSqlResubmissionMode resubmissionMode) {
+        Preconditions.checkNotNull(resubmissionMode, "resubmissionMode");
```

Comment:
same here with the message + maybe worth using `@Nonnull` on parameter and in method below - on the return type